dist-tracing/src/trace.rs

```diff
@@ -0,0 +1,255 @@
+use chrono::{DateTime, Utc};
```

Comment:
this whole module feels honeycomb-specific...I believe other distributed tracing systems have their own notions of trace and span IDs that might have different representations, and the format these are parsed from feels honeycomb-specific. what do you think about replacing the implementation in the `dist-tracing` crate with an interface and moving this implementation to the `honeycomb-tracing` crate?

I think the `Telemetry` trait could have associated types defining IDs, something like

```rust
pub trait Telemetry {
    type TraceId: Clone + Hash + Eq + FromStr + fmt::Display;
    type SpanId: Clone + Hash + Eq + FromStr + fmt::Display;
    // ...
}
```
or something?

Reply:
This is almost certainly the right thing to do, but it'll result in a small usability hit - I've been trying to keep the out-of-band functions (`TraceCtx::new_root().record_on_current_span()`, `TraceCtx::current_trace_ctx()`) free of type parameters. 

I suppose it would be fine to export specialized wrapper functions from the `honeycomb-tracing` crate, and to expect any other such crates to do so as well, and the small bit of extra boilerplate would be worth it for the added flexibility.

Reply:
update: implemented this